### ISSUE TITLE
[musicxml] Finale import of ottava (and presumably related directions)

### ISSUE THREAD

**Issue by rpatters1** [musescore/MuseScore#24870]
### Issue type

Import/export issue

### Description with steps to reproduce

A Finale file that looks like this

<img width="228" alt="Screen Shot 2024-09-22 at 9 41 51 AM" src="https://github.com/user-attachments/assets/eb8e744a-3333-40dd-87f7-2a5d3dc40d22">

exports the attached `8va.musicxml` file (zipped for uploading).

The `<octave-shift type="stop">` element appears before the final E, even though in Finale the E will be played back as ottava (and Finale's xml import includes the E in the ottava.)

The grace notes are not included in the ottava (a shortcoming of Finale). Thus the MuseScore import looks like this:

<img width="191" alt="Screen Shot 2024-09-22 at 9 47 30 AM" src="https://github.com/user-attachments/assets/573113ff-1b8b-4183-8d51-7e6e776dfc7b">

I can improve the import by hand-editing `8va.musicxml` (see attachment) and get the following import into MuseScore:

<img width="217" alt="Screen Shot 2024-09-22 at 9 51 04 AM" src="https://github.com/user-attachments/assets/f5bb8366-32f6-40f6-8ba8-aa8174110492">

The hand edits were:

- Increase the octave number of each grace note by 1.
- Move the `<octave-shift type="down">` before the first grace note
- Move the `<octave-shift type="stop">` after the E.

The issue is, given the new placement of `<octave-shift type="down">`, should the 8va line not visually start at the 1st grace note?

Beyond this, is Finale's export doing it wrong? If so, can/should we compensate? I am considering writing a massager script for Finale xml files to fix some of the export issues. (The grace note issue would be one.) But I'd like to know of this group thinks it makes any sense for MuseScore's import to compensate for any of these issues.

FWIW: this issue is even worse if the 8va ends on the D. Then Finale's exported 8va contains no notes and MuseScore apparently ignores it.

### Supporting files, videos and screenshots

[8va.musicxml.zip](https://github.com/user-attachments/files/17090130/8va.musicxml.zip)
[8va-edited.musicxml.zip](https://github.com/user-attachments/files/17090148/8va-edited.musicxml.zip)

### What is the latest version of MuseScore Studio where this issue is present?

4.4.2

### Regression

I was unable to check

### Operating system

macOS 12.7.6

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by oktophonie** [comment]
There is some work in progress re ottava import at https://github.com/musescore/MuseScore/pull/24776. It may make sense to incorporate any changes we make regarding this issue there. @miiizen is bound to have thoughts.

Due to unfortunate design decisions long ago, MuseScore treats grace notes as attachments to 'real' notes, not as actual notes in themselves. This causes unending problems with layout and playback and until that is solved (https://github.com/musescore/MuseScore/issues/18484) I'm not sure it's worth tinkering around the issue. For now, it means you can't actually visually attach an 8va to a grace note in MuseScore (nor a clef, nor a dynamic, or lyric...)

**Comment by rpatters1** [comment]
Lol, I guess MuseScore inherited Finale's bad design around grace notes.

**Comment by rpatters1** [comment]
I'm learning MusicXML as I go, so I will mention the obvious in the unlikely case it isn't obvious. It is easy to tell if a `.musicxml` was exported from Finale.

**Comment by Jojo-Schmitz** [comment]
Yes it is, and the MusicXML import code has several spwecial cases for Finale already (and for Sibelius and for ...)

**Comment by rpatters1** [comment]
As I suspected in the original post, it turns out the `stop` type `<bracket>` directions have the same problem as those for `<octave-shift>`. There is a disagreement between Finale (on one side) and MuseScore and Dorico (on the other) about how to end them. Finale thinks they should extend *to* the final note while M & D think they should *include* the final note.

The MuseScore import will be much improved if (for Finale-produced `musicxml` files) we extend them by a note/chord. I am attaching a musicxml file with a "fake" ottava symbol that is actually just a text line. It looks like this in Finale.

<img width="376" alt="Screen Shot 2024-10-08 at 3 52 14 PM" src="https://github.com/user-attachments/assets/319faca3-2a42-471b-8419-34160605f8aa">

In Finale one uses these instead of a "real" ottavas because they allow the line to be non-horizontal. (Finale's "real" ottavas can only be flat.) To be clear, the same issue exists for many other types of text lines, from rit/accel to performance instructions, etc.

I'm still looking to see if other `<direction>` types have the same or similar issue.

[8va-fake.musicxml.zip](https://github.com/user-attachments/files/17299190/8va-fake.musicxml.zip)
